.github/workflow/multi-arch-build.yaml

```diff
@@ -0,0 +1,208 @@
+                            skopeo --storage-driver=vfs version)"
+          echo "$VERSION_OUTPUT"
+          VERSION=$(grep -Em1 '^Version: ' <<<"$VERSION_OUTPUT" | awk '{print $2}')
+          test -n "$VERSION"
+          echo "::set-output name=version::${VERSION}"
+
+      - name: Generate skopeo reg. image FQIN(s)
+        id: skopeo_reg
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            # The `skopeo version` in image just built
+            VERSION='v${{ steps.sniff_test.outputs.version }}'
+            # workaround vim syntax-highlight bug: '
+            # Image tags previously pushed to quay
+            ALLTAGS=$(skopeo list-tags \
+                      docker://$SKOPEO_QUAY_REGISTRY/stable | \
+                      jq -r '.Tags[]')
+
+            # New image? Push quay.io/skopeo/stable:vX.X.X and :latest
+            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
+              # Assume version-tag is also the most up to date (i.e. "latest")
+              FQIN="$SKOPEO_QUAY_REGISTRY/stable:$VERSION,$SKOPEO_QUAY_REGISTRY/stable:latest"
+            else # Not a new version-tagged image
+              # Assume other contents changed, so this is the "new" latest.
+              FQIN="$SKOPEO_QUAY_REGISTRY/stable:latest"
+            fi
+          elif [[ "${{ matrix.source }}" == 'testing' ]]; then
+            # Assume some contents changed, always push latest testing.
+            FQIN="$SKOPEO_QUAY_REGISTRY/testing:latest"
+          elif [[ "${{ matrix.source }}" == 'upstream' ]]; then
+            # Assume some contents changed, always push latest upstream.
+            FQIN="$SKOPEO_QUAY_REGISTRY/upstream:latest"
+          else
+            echo "::error::Unknown matrix item '${{ matrix.source }}'"
+            exit 1
+          fi
+          echo "::warning::Pushing $FQIN"
+          echo "::set-output name=fqin::${FQIN}"
+          echo '::set-output name=push::true'
+
+      # This is substantially the same as the above step, except the
+      # $CONTAINERS_QUAY_REGISTRY is used and the "testing"
+      # flavor is never pushed.
+      - name: Generate containers reg. image FQIN(s)
+        if: matrix.source != 'testing'
+        id: containers_reg
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            VERSION='v${{ steps.sniff_test.outputs.version }}'
+            # workaround vim syntax-highlight bug: '
+            ALLTAGS=$(skopeo list-tags \
+                      docker://$CONTAINERS_QUAY_REGISTRY/skopeo | \
+                      jq -r '.Tags[]')
+
+            # New image? Push quay.io/containers/skopeo:vX.X.X and :latest
+            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
+              FQIN="$CONTAINERS_QUAY_REGISTRY/skopeo:$VERSION,$CONTAINERS_QUAY_REGISTRY/skopeo:latest"
+            else # Not a new version-tagged image, but contents may be updated
+              FQIN="$CONTAINERS_QUAY_REGISTRY/skopeo:latest"
```

Comment:
I'm confused by this, for the upstream and testing versions, we should only be pushing a `:latest` tag.  We don't have to worry about adding a `:v1.2.0` or whatever version tag.

Reply:
This entire block excludes `testing` with `if: matrix.source != 'testing'`...But, I think your confusion is justified as it seems this logic will overwrite `latest` from "stable" and again with "upstream".  I'll need to fix this here and in buildah/podman.

Reply:
Follow-up fixes:
* https://github.com/containers/podman/pull/10414
* https://github.com/containers/buildah/pull/3247